packages/paste-core/components/textarea/__tests__/textarea.test.tsx

```diff
@@ -73,6 +78,48 @@ describe('TextArea render', () => {
   });
 });
 
+describe('Multiple textareas', () => {
+  it('should render 2 hidden textareas if there are 2 textareas', () => {
+    render(<MultipleTextareas />);
+    const pushButton = screen.getByRole('button', {name: 'Push textarea'});
+
+    expect(screen.getAllByRole('textbox', {hidden: true})).toHaveLength(2);
+
+    fireEvent.click(pushButton);
+    expect(screen.getAllByRole('textbox', {hidden: true})).toHaveLength(4);
+  });
+
+  it('should remove 1 hidden textarea if 1 textarea is removed', () => {
+    render(<MultipleTextareas />);
+
+    const pushButton = screen.getByRole('button', {name: 'Push textarea'});
+    const popButton = screen.getByRole('button', {name: 'Pop textarea'});
+
+    expect(screen.getAllByRole('textbox', {hidden: true})).toHaveLength(2);
+
+    fireEvent.click(pushButton);
+    expect(screen.getAllByRole('textbox', {hidden: true})).toHaveLength(4);
+
+    fireEvent.click(popButton);
+    expect(screen.getAllByRole('textbox', {hidden: true})).toHaveLength(2);
+  });
+
+  it('should remove all hidden textarea if all textareas are removed', () => {
+    render(<MultipleTextareas />);
+
+    const pushButton = screen.getByRole('button', {name: 'Push textarea'});
+    const toggleButton = screen.getByRole('button', {name: 'Toggle textarea visibility'});
+
+    expect(screen.getAllByRole('textbox', {hidden: true})).toHaveLength(2);
+
+    fireEvent.click(pushButton);
+    expect(screen.getAllByRole('textbox', {hidden: true})).toHaveLength(4);
+
+    fireEvent.click(toggleButton);
+    expect(screen.queryAllByRole('textbox', {hidden: true})).toHaveLength(0);
```

Comment:
quibble: we can merge these all into one `it` test to avoid having to re-render the same `<MultipleTextareas>` several times, since these tests follow a natural order. Test can be re-titled to `handles adding and removing multiple text areas`

Reply:
Updated.